grr/server/grr_response_server/flows/general/osquery.py

```diff
@@ -152,7 +154,13 @@ def _GetPathSpecsToCollect(
     file_names = []
     for osquery_result in responses:
       for column in self.args.file_collection_columns:
-        file_names.extend(osquery_result.table.Column(column))
+        try:
+          values_in_column = osquery_result.table.Column(column)
+          file_names.extend(values_in_column)
+        except KeyError:
+          self._UpdateProgressWithError(
+              f"No such column '{column}' to collect files from.")
+          raise
```

Comment:
Scope the `try` block only to the part where the exception can happen (i.e. move the `_.extend(_)` part out of it).

Reply:
`Column(1)` returns an iterator, so the exception actually occurs when I call `extend`: https://github.com/simstoykov/grr/blob/final_polish/grr/core/grr_response_core/lib/rdfvalues/osquery.py#L56

Should I move `values_in_column = osquery_result.table.Column(column)` out of it above, or maybe inline `values_in_column`?

Reply:
Ah, these iterators... Personally, I would inline `values_in_column` but you can leave it as it is if you prefer it.